tests/wrong_coverage.py

```diff
@@ -0,0 +1,120 @@
+import re
+import subprocess
+import sys
+from pathlib import Path
+from tempfile import NamedTemporaryFile
+
+from pydantic import BaseModel, Field
+from rich.console import Console
+
+
+def main(exclude_comment: str = 'pragma: no cover') -> int:
+    with NamedTemporaryFile(suffix='.json') as coverage_json:
+        with NamedTemporaryFile(mode='w', suffix='.toml') as config_file:
+            config_file.write(f"[tool.coverage.report]\nexclude_lines = ['{exclude_comment}']\n")
+            config_file.flush()
+            p = subprocess.run(
+                ['uv', 'run', 'coverage', 'json', f'--rcfile={config_file.name}', '-o', coverage_json.name],
+                stdout=subprocess.PIPE,
+                stderr=subprocess.STDOUT,
+            )
+            if p.returncode != 0:
+                print(f'Error running coverage:\n{p.stdout.decode()}', file=sys.stderr)
+                return p.returncode
+
+        r = CoverageReport.model_validate_json(coverage_json.read())
+
+    blocks: list[str] = []
+    total_lines = 0
+    cwd = Path.cwd()
+    for file_name, file_coverage in r.files.items():
+        # Find lines that are both excluded and executed
+        common_lines = sorted(set(file_coverage.excluded_lines) & set(file_coverage.executed_lines))
+
+        if not common_lines:
+            continue
+
+        code_analysise: CodeAnalyzer | None = None
+
+        def add_block(start: int, end: int):
+            nonlocal code_analysise, total_lines
+
+            if code_analysise is None:
+                code_analysise = CodeAnalyzer(file_name)
+
+            if not code_analysise.all_block_openings(start, end):
+                b = str(start) if start == end else f'{start}-{end}'
+                if not blocks or blocks[-1] != b:
+                    total_lines += end - start + 1
+                    blocks.append(f'  [link=file://{cwd / file_name}]{file_name} {b}[/link]')
```

Comment:
would be really nice to link to the specific line here, but adding `:{start}` here means the link isn't clickable in zed.